task_sdk/src/airflow/sdk/api/client.py

```diff
@@ -134,6 +135,13 @@ def defer(self, id: uuid.UUID, msg):
         # Create a deferred state payload from msg
         self.client.patch(f"task-instances/{id}/state", content=body.model_dump_json())
 
+    def reschedule(self, id: uuid.UUID, msg):
+        """Tell the API server that this TI has been reschduled."""
+        body = TIRescheduleStatePayload(**msg.model_dump(exclude_unset=True))
```

Comment:
What is msg here? It should probably have a type, or if it's just the datetime pass that in directly and create the message type in this function (similar to how we do for `heartbeat` and `finish` methods. We should likely have done `defer` differently too. Maybe.)

Reply:
Its of `RescheduleTask` class. The reason we just passed it down to this api endpoint is so that we do not need to worry about unpacking the object in the `handle_requests`. 
```
        elif isinstance(msg, RescheduleTask):
            self._terminal_state = IntermediateTIState.UP_FOR_RESCHEDULE
            self.client.task_instances.reschedule(self.id, msg)
            resp = None
```

But a type can be added for sure. Let me do that

Reply:
Yeah, I go back and forth on it.

`RescheduleTask` is an easy one because it currently has only one attribute: `reschedule_date`, which is needed from `Task Runner` -> `Supervisor`, so it can be passed an explicit argument.

But for something like defer it has `classpath`, `trigger_kwargs`, `next_method` and `trigger_timeout`, so we just need to decide whether we want to keep client code decoupled. The downside is some code duplication

We don't strictly need to do that, as the `payload` expected by various endpoints, and therefore for client, are all in `_generated.py` which are inherited in `comms.py` anyway. So there is already a coupling of "comm messages" between Supervisor & Task Runner with the Some of the Payloads.

----

But regardless of that, yes we should have the typehints.

Reply:
Typehints are already added. Maybe in a follow up we can decide and perform some cleanup.

Reply:
Resolving for now, will revisit during cleanup